st2api/st2api/controllers/v1/actionviews.py

```diff
@@ -98,32 +95,36 @@ class OverviewController(resource.ContentPackResourceController):
 
     include_reference = True
 
-    @jsexpose(arg_types=[str])
     def get_one(self, ref_or_id):
         """
             List action by id.
 
             Handle:
                 GET /actions/views/overview/1
         """
-        action_api = super(OverviewController, self)._get_one(ref_or_id)
-        return self._transform_action_api(action_api)
+        resp = super(OverviewController, self)._get_one(ref_or_id)
+        action_api = resp.json
+        result = self._transform_action_api(action_api)
+        resp.json = result
+        return resp
 
-    @jsexpose(arg_types=[str])
     def get_all(self, **kwargs):
         """
             List all actions.
 
             Handles requests:
                 GET /actions/views/overview
         """
-        action_apis = super(OverviewController, self)._get_all(**kwargs)
-        return map(self._transform_action_api, action_apis)
+        resp = super(OverviewController, self)._get_all(**kwargs)
+        action_apis = resp.json
+        result = map(self._transform_action_api, action_apis)
+        resp.json = result
+        return resp
 
     @staticmethod
     def _transform_action_api(action_api):
-        action_id = action_api.id
-        action_api.parameters = ParametersViewController._get_one(action_id).get('parameters')
+        action_id = action_api['id']
```

Comment:
```class.to_json()``` is already implemented by means of `__json__()` method. Almost the entire purpose of `json_encode()` utility is to properly unwrap it.

As far as what controller should return, well, yes, you're right and we should probably keep using models till the very end. I might be just trying to express my frustration at the fact that we as a team sometimes forget what exactly our API is returning to the user and that it is our job to design api model properly so it could be serialized to json without additional fiddle on Router side.